magento/pwa-studio
# [PWA-653] React Refresh

## Description

Implement React Refresh using [@pmmmwh/react-refresh-webpack-plugin](https://github.com/pmmmwh/react-refresh-webpack-plugin)

## Related Issue





Closes #2391.

## Acceptance 



### Verification Stakeholders
@dani97 
@Jordaneisenburger 

### Specification



### Verification Steps


1. Ensure that you have a valid custom origin setup
2. Run `yarn install`
3. Run `yarn watch:venia`
4. Open the browser and code editor in a side by side environment, edit a component such as `Main.js` in `venia-concept`, adding `<div>Hello</div>` above the header, save and see the changes applied.
5. Edit the component to say `<div>Hello there</div>` and see the changes applied without a full page refresh.
6. Generate an error by not closing the tag `<div>Hello there<div>`, see error message **displayed.**


## Screenshots / Screen Captures (if appropriate)
![React Refresh](https://recordit.co/CwdZpYC2x5.gif)

Error Messaging
![image](https://user-images.githubusercontent.com/44360092/89524385-62e25b00-d7dc-11ea-92e2-86afb9f7e60f.png)



## Checklist

* I have updated the documentation accordingly, if necessary.
